kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeDownloadPanel.js

```diff
@@ -54,9 +54,11 @@ define([
             const lastUpdateTime = this.downloadSpecCache['lastUpdateTime'];
             if (lastUpdateTime) {
                 this.render();
+                return Promise.resolve(this);
```

Comment:
In some cases, this would return a Promise at startup. Now it ALWAYS returns a Promise for better flow control.